### PR TITLE
Added basic blink example for Esp32-C6

### PR DESCRIPTION
**Pull request by allexoK**
Hello,

When learning about embedded-Swift, I was surprised that there was no basic blink example (only the LED strip one). So, I decided to create one.

I used the ESP32-C6-Bug board, but it doesn’t really matter as long as there is a normal LED on-board. Alternatively, an external LED can be connected.

### CODE REVIEW

File: esp32-led-blink-sdk/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cmake_minimum_required(VERSION 3.16)
```
Comment (rauhul) [inline esp32-led-blink-sdk/CMakeLists.txt:1]: Could you increase this to 3.29 and ensure everything still works?

File: esp32-led-blink-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# esp32-led-strip-sdk
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:1]: I think this name is copy-pasted, could you update it?

File: esp32-led-blink-sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+This example demonstrates how to integrate with the ESP-IDF SDK via CMake and how to use the standard GPIO library to control LED from Swift. This example is specifically made for the RISC-V MCUs from ESP32 (the Xtensa MCUs are not currently supported by Swift).
+
+![Led on](https://github.com/allexoK/swift-embedded-examples/assets/50021377/c40d4b3a-670b-4743-a8ae-a90c90b905b4)
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:5]: These links appear to point out of tree, could you fix them?

File: esp32-led-blink-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+- Before trying to use Swift with the ESP-IDF SDK, make sure your environment works and can build the provided C/C++ sample projects, in particular:
+  - Try building and running the "get-started/blink" example from ESP-IDF written in C.
+
+## Building
+
+- Make sure you have a recent nightly Swift toolchain that has Embedded Swift support.
+- If needed, run export.sh to get access to the idf.py script from ESP-IDF.
+- Specify the nightly toolchain to be used via the `TOOLCHAINS` environment variable and the target board type by using `idf.py set-target`.
+``` console
+$ cd esp32-led-strip-sdk
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:22]: copied folder name

File: esp32-led-blink-sdk/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+$ export TOOLCHAINS=...
+$ . <path-to-esp-idf>/export.sh
+$ idf.py set-target esp32c6
+$ idf.py build
+```
+
+## Running
+
+- Connect the Esp32-C6-Bug board(Or any other board with integrated LED on GPIO pin 8) over a USB cable to your Mac. Alternatively you can just connect external LED to GPIO pin 8 on any other board.
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:31]: could you adjust the formatting here slightly: "...board (or any..."

File: esp32-led-blink-sdk/main/Led.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Modified by Karavaev Aleksei
```
Comment (rauhul) [inline esp32-led-blink-sdk/main/Led.swift:5]: Please remove this line all files should follow the same header

File: esp32-led-blink-sdk/main/Main.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Modified by Karavaev Aleksei
```
Comment (rauhul) [inline esp32-led-blink-sdk/main/Main.swift:5]: same here

File: esp32-led-blink-sdk/main/BridgingHeader.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors.
```
Comment (rauhul) [inline esp32-led-blink-sdk/main/BridgingHeader.h:5]: Could you make sure to use 2024 as the copyright date on these new additions

File: esp32-led-blink-sdk/main/Led.swift [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+struct Led {
+  var ledPin: gpio_num_t
+  init(gpioPin: Int) {
+    ledPin = gpio_num_t(Int32(gpioPin))
+
+    guard gpio_reset_pin(ledPin) == ESP_OK else {
+      fatalError("cannot reset led")
+    }
+
+    guard gpio_set_direction(ledPin,GPIO_MODE_OUTPUT) == ESP_OK else {
```
Comment (rauhul) [inline esp32-led-blink-sdk/main/Led.swift:23]: nit: add a space after the comma

File: esp32-led-blink-sdk/main/Main.swift [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+// Copyright (c) 2023 Apple Inc. and the Swift project authors.
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+// The code will blink a led attached to GPIO8, to change the led pin change the Led(gpioPin: 8) to something else
```
Comment (rauhul) [inline esp32-led-blink-sdk/main/Main.swift:13]: nit: wrap to 80 characters

File: esp32-led-blink-sdk/README.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+- Connect RX pin of USB-UART converter to TX0 pin of your board if you need serial ouput.(You may also need to connect GND converter pin to the GND pin of the board)
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:32]: same here could you add a space after output and remove the parens "...output. You may..."

**Review by rauhul — CHANGES_REQUESTED** [review]
Thanks for the PR there are some hopefully simple changes we need to make before merging this addition :)

Lastly could you also update the README at the repo root to list this example

File: esp32-led-blink-sdk/README.md (hunk 2 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: sure, done

File: esp32-led-blink-sdk/README.md (hunk 3 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: I created the pics directory in the project folder and uploaded the pictures there, is it fine?

File: esp32-led-blink-sdk/README.md (hunk 4 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: fixed

File: esp32-led-blink-sdk/README.md (hunk 5 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: done

File: esp32-led-blink-sdk/main/Main.swift (hunk 7 above)
Comment (allexoK) [reply esp32-led-blink-sdk/main/Main.swift]: sure, fixed

File: esp32-led-blink-sdk/main/BridgingHeader.h (hunk 8 above)
Comment (allexoK) [reply esp32-led-blink-sdk/main/BridgingHeader.h]: changed in all files of the project

File: esp32-led-blink-sdk/main/Led.swift (hunk 9 above)
Comment (allexoK) [reply esp32-led-blink-sdk/main/Led.swift]: fixed

File: esp32-led-blink-sdk/main/Main.swift (hunk 10 above)
Comment (allexoK) [reply esp32-led-blink-sdk/main/Main.swift]: fixed

File: esp32-led-blink-sdk/README.md (hunk 11 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: done

**Comment by allexoK** [comment]
Hi @rauhul, thanks for review.
I think I fixed everything. I was not sure to which table to add the example(Catalog of Examples or Community Examples). 'Community examples' table points to the external repos, so I decided to go with 'Catalog of Examples'.
Let me know if any other change is required :)

File: esp32-led-blink-sdk/README.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
+# esp32-led-blink-sdk
+
+This example demonstrates how to integrate with the ESP-IDF SDK via CMake and how to use the standard GPIO library to control LED from Swift. This example is specifically made for the RISC-V MCUs from ESP32 (the Xtensa MCUs are not currently supported by Swift).
+
+![Led on](pics/ledon.jpg)
```
Comment (rauhul) [inline esp32-led-blink-sdk/README.md:5]: last nit: could you change this path to `assets/images/...`

**Review by rauhul — CHANGES_REQUESTED** [review]
@allexoK one last change and were good to go!

**Review by rauhul — APPROVED** [review]
fantastic, thank you for the contribution!

File: esp32-led-blink-sdk/README.md (hunk 12 above)
Comment (allexoK) [reply esp32-led-blink-sdk/README.md]: Sure, changed it :)

**Pull request merged**